portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -29,21 +52,45 @@
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
```

Comment:
Rather than create a new DatastoreService object with every request, consider creating one as a class member, initializing it in the constructor and reusing it. Creating new versions of services can be slow. You could do something like this:

```
public class DataServlet extends HttpServlet {

  protected DatastoreService datastore;

  public DataServlet() {
    super();  // Don't forget to always call the parent constructor!
    datastore = DatastoreServiceFactory.getDatastoreService();
  }
```

then inside doGet and doPost you could refer to the class member instead of creating a new one. Same with the Gson object you're creating -- it can be reused instead of recreated (see https://github.com/google/gson/blob/master/UserGuide.md#using-gson).

Reply:
Could you explain the super( ) part?